balboa-agent/src/main/scala/com/socrata/balboa/agent/BalboaAgent.scala

```diff
@@ -52,15 +40,21 @@ object BalboaAgent extends App with Config with LazyLogging {
   logger info s"Starting Balboa Agent.  Consuming metrics from ${dataDir.getAbsolutePath}.  " +
     s"AMQ Server: $activemqServer, AMQ Queue: $activemqQueue $additionalInfo".trim
 
-  val mc = (activemqUser, activemqPassword) match {
-    case (Some(user), Some(password)) => new MetricConsumer(
-      dataDir, MetricJmsQueue.getInstance(user, password, activemqServer, activemqQueue))
-    case _ => new MetricConsumer(dataDir, MetricJmsQueue.getInstance(activemqServer, activemqQueue))
+  val amqConnection = (activemqUser, activemqPassword) match {
+    case (Some(u), Some(p)) => ActiveMQConnection.makeConnection(u, p, activemqServer)
+    case _ => ActiveMQConnection.makeConnection(activemqServer)
   }
-  val future: ScheduledFuture[_] = scheduler.scheduleWithFixedDelay(mc, INITIAL_DELAY, period, TimeUnit.MILLISECONDS)
+  logger.debug(s"Activemq Connection $amqConnection")
+  val mc = new MetricConsumer(dataDir, new MetricJmsQueueNotSingleton(amqConnection, activemqQueue))
+
+  logger.debug(s"Metric Consumer $mc")
+  // TODO (Pre req: Java 8) Use Java Duration class.
+  val future: ScheduledFuture[_] = scheduler.scheduleWithFixedDelay(mc, initialDelay(), interval(), TimeUnit.MILLISECONDS)
 
   Runtime.getRuntime.addShutdownHook(new Thread() {
     override def run(): Unit = {
+      logger info s"Shutting down Balboa Agent.  Attempt to cancel and future subsequent runs."
```

Comment:
Don't think this has to be a Scala string.

If you put an 'attempt' message in the log, there should be corresponding success and failure messages at the right place. Otherwise, as a log reader, I'm left wondering if something crashed while 'attempting'.